src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,413 @@
+"""
+    FileReader
+
+This module coordinates reading of input data from NetCDF files,
+and allows us to read in data containing a time component.
+It also includes regridding and temporal interpolations
+of this data.
+
+This is based on ClimaCoupler.jl's BCReader and TimeManager modules.
+"""
+module FileReader
+
+using ClimaComms
+using ClimaCore: Fields, Spaces
+using Dates
+using JLD2
+import CFTime
+
+using ClimaLSM.Regridder
+
+export to_datetime,
+    datetime_to_strdate,
+    FileInfo,
+    FileState,
+    SimInfo,
+    PrescribedData,
+    prescribed_data_init,
+    read_data_fields!,
+    next_date_in_file,
+    interpolate_data
+
+
+"""
+    FileInfo
+
+Stores information about the current data being read in from a file.
+
+# Inputs:
+- regrid_dirpath::String     # directory for storing files used in regridding
+- outfile_root::String       # root for files generated when reading data from input file
+- varname::String            # name of the variable we're reading from the input file
```

Comment:
We should also note somewhere that right now we are assuming this variable is a scalar.

One other thought is to note in the doc strings that by date we mean datetime but leave it as "date" for brevity.

Reply:
Which variable are we assuming is a scalar? `varname`?

Reply:
I elaborated the docstrings for `varname` and `all_dates` to make them more clear